executor/adapter.go

```diff
@@ -232,17 +233,38 @@ func (a *ExecStmt) PointGet(ctx context.Context, is infoschema.InfoSchema) (*rec
 		return nil, err
 	}
 	a.Ctx.GetSessionVars().StmtCtx.Priority = kv.PriorityHigh
-	b := newExecutorBuilder(a.Ctx, is)
-	exec := b.build(a.Plan)
-	if b.err != nil {
-		return nil, b.err
+
+	// try to reuse point get executor
+	var pointExecutor *PointGetExecutor
+
+	if a.PsStmt.Executor != nil {
+		exec, ok := a.PsStmt.Executor.(*PointGetExecutor)
+		if !ok {
+			logutil.Logger(ctx).Error("invalid executor type, not PointGetExecutor for point get path")
+			a.PsStmt.Executor = nil
+		} else {
+			// CachedPlan type is already checked in last step
+			exec.idxVals = a.PsStmt.PreparedAst.CachedPlan.(*plannercore.PointGetPlan).IndexValues
```

Comment:
Can we extract this into a function and reuse the function to build the executor?
So when we modify the executor builder in the future, we don't need to change two places.

Reply:
@coocood  good idea, do we need to abstract a `Reset` like interface for `Executor` so this can be reused like physical plan `RebuildRange`,  which may improve some simple benchmark scenario like point execution ?

Reply:
Not for now, it's very error-prone, so we intentionally don't reuse it.